# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../jbigkit/shared-lib.patch
# Copyright (C) 2010 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

--- jbigkit/libjbig/Makefile.orig	2008-08-30 19:20:52.000000000 +0200
+++ jbigkit/libjbig/Makefile	2010-08-13 11:05:57.632003425 +0200
@@ -7,7 +7,13 @@
 # Options for the compiler: A high optimization level is suggested
 CFLAGS = -g -O -Wall -ansi -pedantic # --coverage
 
-all: libjbig.a tstcodec tstcodec85
+all: static shared tstbin
+
+static: libjbig.a libjbig85.a
+
+shared: libjbig.so libjbig85.so
+
+tstbin: tstcodec tstcodec85
 
 tstcodec: tstcodec.o jbig.o jbig_ar.o
 	$(CC) $(CFLAGS) -o tstcodec $+
@@ -25,6 +31,12 @@
 	ar rc libjbig85.a jbig85.o jbig_ar.o
 	-ranlib libjbig85.a
 
+libjbig.so: jbig.o jbig_ar.o
+	$(CC) $(CFLAGS) -shared jbig.o jbig_ar.o -o libjbig.so
+
+libjbig85.so: jbig85.o jbig_ar.o
+	$(CC) $(CFLAGS) -shared jbig85.o jbig_ar.o -o libjbig85.so
+
 jbig.o: jbig.c jbig.h jbig_ar.h
 jbig85.o: jbig85.c jbig85.h jbig_ar.h
 jbig_ar.o: jbig_ar.c jbig_ar.h
